compiler/src/iree/compiler/Dialect/TensorExt/IR/TensorExtOpFolders.cpp

```diff
@@ -313,21 +313,26 @@ struct BubbleUpOrdinalOp : public OpRewritePattern<DispatchWorkloadOrdinalOp> {
 
 } // namespace
 
-/// Fold away following sequence of `iree_tensor_ext.dispatch.workload.ordinal`.
-///
-/// ```mlir
-/// %1 = iree_tensor_ext.dispatch.workload.ordinal %0 2
-/// %2 = iree_tensor_ext.dispatch.workload.ordinal %1 2
-/// ```
-///
-/// This can happen when the operands get deduped.
 OpFoldResult DispatchWorkloadOrdinalOp::fold(FoldAdaptor operands) {
+  // If the operand being annotated is a constant then just fold to it as
+  // there's no longer any relation to the captured workload.
+  if (operands.getOperand()) {
+    return operands.getOperand();
+  }
+
+  // Fold away following sequence ordinal ops:
+  // ```mlir
+  // %1 = iree_tensor_ext.dispatch.workload.ordinal %0, 2
+  // %2 = iree_tensor_ext.dispatch.workload.ordinal %1, 2
+  // ```
+  // This can happen when the operands get deduped.
```

Comment:
I'm not a fan of having markdown style block in comments, because it does not provide values. They can just be blank.

```suggestion
  // Fold away following sequence ordinal ops:
  //
  //   %1 = iree_tensor_ext.dispatch.workload.ordinal %0, 2
  //   %2 = iree_tensor_ext.dispatch.workload.ordinal %1, 2
  //
  // This can happen when the operands get deduped.
```